pinot-core/src/test/java/com/linkedin/pinot/core/realtime/impl/dictionary/ConcurrentReadWriteDictionaryTest.java

```diff
@@ -79,44 +171,60 @@ public void tearDown() {
    * <p>We can assume that we always first get the index of a value, then use the index to fetch the value.
    */
   private class Reader implements Callable<Void> {
+    private final MutableDictionary _dictionary;
+    private Reader(MutableDictionary dictionary) {
+      _dictionary = dictionary;
+    }
 
     @Override
     public Void call()
         throws Exception {
-      for (int i = 0; i < NUM_ENTRIES; i++) {
-        int dictId;
-        do {
-          dictId = _intOnHeapMutableDictionary.indexOf(i + 1);
-        } while (dictId < 0);
-        Assert.assertEquals(dictId, i);
-        Assert.assertEquals(_intOnHeapMutableDictionary.getIntValue(dictId), i + 1);
-
-        // Fetch value by a random existing dictId
-        int randomDictId = RANDOM.nextInt(i + 1);
-        Assert.assertEquals(_intOnHeapMutableDictionary.getIntValue(randomDictId), randomDictId + 1);
+      try {
```

Comment:
So that if we see a throwable, we can reproduce the test case with the right random seed.